core/drivers/pcap.cc

```diff
@@ -47,27 +50,27 @@ int PCAPPort::RecvPackets(queue_t qid, bess::Packet** pkts, int cnt) {
       break;
     }
 
-    if (caplen <= SNBUF_DATA) {
-      // pcap packet will fit in the mbuf, go ahead and copy.
-      bess::utils::CopyInlined(sbuf->append(caplen), packet, caplen, true);
-    } else {
-/* FIXME: no support for chained mbuf for now */
-#if 0
-      /* Try read jumbo frame into multi mbufs. */
-      if (unlikely(pcap_rx_jumbo(sbuf->mbuf.pool,
-              &sbuf->mbuf,
-              packet,
-              caplen) == -1)) {
-        //drop all the mbufs.
-        bess::Packet::Free(sbuf);
-        break;
-      }
-#else
-      RTE_LOG(ERR, PMD, "Dropping PCAP packet: Size (%d) > max size (%d).\n",
-              sbuf->total_len(), SNBUF_DATA);
-      bess::Packet::Free(sbuf);
-      break;
-#endif
+    int copy_len = std::min(caplen, static_cast<int>(sbuf->tailroom()));
+    bess::utils::CopyInlined(sbuf->append(copy_len), packet, copy_len, true);
+
+    packet += copy_len;
+    caplen -= copy_len;
+    bess::Packet* m = sbuf;
+
+    while (caplen > 0) {
+      m->set_next(bess::Packet::Alloc());
+      m = m->next();
+
+      // no headroom needed in chained mbufs
+      m->prepend(m->headroom());
+      m->set_data_len(0);
+      m->set_buffer(0);
+
+      copy_len = std::min(caplen, static_cast<int>(m->tailroom()));
+      bess::utils::CopyInlined(m->append(copy_len), packet, copy_len, true);
```

Comment:
I think a non-inlined copy is good enough here... We already know that the packet is large, so tiny performance gain would not outweigh the increased code size.